### PR TITLE
Meme Generator API & Corpus Script

### PR DESCRIPTION
**Pull request by schmamps**
./corpus.py is a simple script to generate a corpus of text
API handled by mgapi module

### CODE REVIEW

File: mgapi/handler.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        for i, param in enumerate(params):
+            url = self._add_url_param(url, param, params[param])
+
+        return url
+
+    def _add_url_param(self, url, name, val):
+        """ adds a parameter to the URL string """
+        delim = '&' if '?' in url else '?'
+        param = '{}{}={}'.format(delim, name, val)
+        return url + param
```
Comment (selassid) [inline mgapi/handler.py]: [urllib.parse.urlencode()](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode) has already implemented this.

File: mgapi/handler.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        """ String builder for API endpoints """
+        url = 'http://{}/{}'.format(
+            'version1.api.memegenerator.net',
+            action
+        )
+
+        for i, param in enumerate(params):
+            url = self._add_url_param(url, param, params[param])
+
+        return url
```
Comment (selassid) [inline mgapi/handler.py]: [urllib.parse.unsplit()](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlunsplit) has also already implemented this.

File: mgapi/handler.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    def _load_url_as_json(self, url):
+        """ Loads a URL, returns JSON data as an object
+
+        :return json_data
+        """
+        with urllib.request.urlopen(url) as text:
+            data_str = '\n'.join([line.decode('UTF-8') for line in text])
+
+        data_json = json.loads(data_str)
```
Comment (davidselassie) [inline mgapi/handler.py]: I recommend using the [requests](http://docs.python-requests.org/en/master/) Python library whenever doing any HTTP getting. Even the [standard lib docs for urllib.request suggest you just use requests](https://docs.python.org/3/library/urllib.request.html).

Turns out it already has implemented three of your functions here for you. You can jump straight to using [requests.get()](http://docs.python-requests.org/en/master/user/quickstart/#passing-parameters-in-urls) with parameters. Then you don't need `_get_request_url()` or `_add_url_param()`.

**Comment by Kaylotura** [comment]
I'm not entirely certain I follow everything that's going on here, but it looks like this aggregate a corpus of popular memes from popular meme creators at Meme Generator.net. Which is perfect.
I say ship it.

**Comment by coreyadkins** [comment]
Looks good to me. Ship it!

**Pull request merged**